extensions/amp-a4a/0.1/refresh-manager.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {analyticsForDoc} from '../../../src/analytics';
+import {timerFor} from '../../../src/services';
+import {
+  getEnclosingContainerTypes,
+  ValidAdContainerTypes,
+} from '../../../ads/google/a4a/utils';
+import {user} from '../../../src/log';
+
+/**
+ * visibilePercentageMin - The percentage of pixels that need to be on screen
+ *   for the creative to be considered "visible".
+ * totalTimeMin - The total amount of time, in milliseconds, that the creative
+ *   must be on screen in order to be considered "visible".
+ * continuousTimeMin - The amount of continuous time, in milliseconds, that the
+ *   creative must be on screen for in oreder to be considered "visible".
+ *
+ * @typedef {{
+ *   visiblePercentageMin: number,
+ *   totalTimeMin: number,
+ *   continuousTimeMin: number,
+ * }}
+ */
+export let RefreshConfig;
+
+export const MIN_REFRESH_INTERVAL = 3;
+export const DATA_ATTR_NAME = 'data-enable-refresh';
+export const METATAG_NAME = 'amp-ad-enable-refresh';
+
+const TAG = 'AMP-AD';
+
+export class RefreshManager {
+
+  /**
+   * @param {!./amp-a4a.AmpA4A} a4a The AmpA4A instance to be refreshed.
+   * @param {!RefreshConfig} config
+   */
+  constructor(a4a, config) {
+
+    /** @const @private {!./amp-a4a.AmpA4A} */
+    this.a4a_ = a4a;
+
+    /** @const @private {!Window} */
+    this.win_ = a4a.win;
+
+    /** @const @private {!Element} */
+    this.element_ = a4a.element;
+
+    /** @const @private {string} */
+    this.adType_ = this.element_.getAttribute('type');
```

Comment:
Added toLowerCase. getAttribute() always returns a string, so there's no need to assert that it's a string.